org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerPlugin.java

```diff
@@ -31,10 +45,53 @@ public class LanguageServerPlugin extends AbstractUIPlugin {
 	public LanguageServerPlugin() {
 	}
 
+	public static class WindowListener implements IPartListener2 {
```

Comment:
Can you please give explicit name here, such as `AttachLSP4EContextPartListener` ?